### DISCUSSION TITLE
How to install only dev dependancies ?

### DISCUSSION THREAD

**Discussion by aahnik** [python-poetry/poetry#3970] [Q&A] [answered]
Along with dev dependancy, is there a way to specify docs dependencies ?

for example, `poetry add mkdocs --doc`

?

I want to install only dev dependencies, while building docs using github actions. How to do that ?

i see

```
  --no-dev               Do not install the development dependencies.

```

is there something `--only-dev` or `--only-doc` ?

**Comment by finswimmer** [comment, marked as answer]
Hello @aahnik 

a `--dev-only` option is already implemented (https://github.com/python-poetry/poetry/pull/2917) and will be available from poetry 1.2 on.

Something like `poetry add --doc mkdocs` will be available once the dependency groups feature (https://github.com/python-poetry/poetry/issues/1644) is implemented. This is also planed for poetry 1.2.

fin swimmer

**Reply by aahnik** [reply]
wow! thanks a lot. these features are great. waiting for the update.

**Reply by cwittlinger** [reply]
Is there a roadmap when 1.2 is comming?
